packages/perseus/src/widgets/categorizer/categorizer.tsx

```diff
@@ -15,31 +15,42 @@ import mediaQueries from "../../styles/media-queries";
 import sharedStyles from "../../styles/shared";
 import {getPromptJSON as _getPromptJSON} from "../../widget-ai-utils/categorizer/categorizer-ai-utils";
 
-import type {Widget, WidgetExports, WidgetProps} from "../../types";
+import type {
+    ChangeFn,
+    UniversalWidgetProps,
+    Widget,
+    WidgetExports,
+} from "../../types";
 import type {CategorizerPromptJSON} from "../../widget-ai-utils/categorizer/categorizer-ai-utils";
 import type {
     PerseusCategorizerWidgetOptions,
     PerseusCategorizerUserInput,
     CategorizerPublicWidgetOptions,
 } from "@khanacademy/perseus-core";
+import type {PropsFor} from "@khanacademy/wonder-blocks-core";
 
-type Props = WidgetProps<RenderProps> & {
-    values: ReadonlyArray<string>;
-};
+interface InternalProps extends UniversalWidgetProps {
```

Comment:
I have a small reaction to renaming this to `InternalProps`. I realize that's the usage of this type, but it feels a bit like we're breaking convention (a vast majority of code both inside Perseus and in other codebases simply refers to this type as `Props`). 

I'm undecided which way I land but thought I'd comment in case other's feel more strongly.

Reply:
For metrics, in Perseus, we the top 5 most popular type name for Props is: 

  * 700 - `Props`
  * 420 - `ChangeProps`
  * 130 - `DefaultProps`
  * 125 - `MafsGraphProps`
  * 66 - `WidgetProps`

In webapp, that numbers are:

 * 5512 - `Props`
 * 493 - `ExportProps`
 * 312 - `OwnProps`
 * 269 - `DefaultProps`
 * 224 - `WithApolloProps`

Reply:
So thinking a bit more about this I think a descriptive name makes sense especially for class components. I think you're addressing the fact that `InternalProps` are what the component sees from inside and `ExternalProps` are what consumers of the component see (mainly differing because of DefaultProps), right? 

Further, for functional components, this can go away because we define `Props` as what the consumer sees and then the defaulting happens when we destructure in the function declaration, relieving us from needing a separately defined type for the internal view of Props. 

TLDR; I'm good with special naming for our class components.